version: "3.9"

services:
  # Remote/production services
  backend:
    image: ${PYGEOAPI_IMAGE:-ghcr.io/yejiyang/pygeoapi-w-global-tsunami-data:latest}
    container_name: backend
    ports:
      - "5000:5000"
    environment:
      - PYGEOAPI_CONFIG=/pygeoapi/local.config.yml
      - PYGEOAPI_OPENAPI=/tmp/openapi.yml
      - CONTAINER_NAME=pygeoapi
      - CONTAINER_HOST=0.0.0.0
      - CONTAINER_PORT=5000
      - PYGEOAPI_SERVER_URL=${PYGEOAPI_SERVER_URL:-http://localhost:5000}

  frontend:
    image: ${FRONTEND_IMAGE:-ghcr.io/yejiyang/global-tsunami-risk-map-frontend:latest}
    container_name: frontend
    ports:
      - "8080:80"
    depends_on:
      - backend

  # Local development services
  backend-local:
    image: geopython/pygeoapi:latest
    container_name: backend
    ports:
      - "5000:5000"
    volumes:
      - ./data:/data
      - ./pygeoapi/pygeoapi-config.yml:/pygeoapi/local.config.yml
    environment:
      - PYGEOAPI_CONFIG=/pygeoapi/local.config.yml
      - PYGEOAPI_OPENAPI=/tmp/openapi.yml
      - CONTAINER_NAME=pygeoapi
      - CONTAINER_HOST=0.0.0.0
      - CONTAINER_PORT=5000
      - PYGEOAPI_SERVER_URL=http://localhost:5000
    profiles:
      - local

  frontend-local:
    image: local/global-tsunami-risk-map-frontend:latest
    container_name: frontend
    ports:
      - "8080:80"
    depends_on:
      - backend-local
    profiles:
      - local
